config/application.php

```diff
@@ -55,23 +55,21 @@
 /**
  * DB settings
  */
-Config::define('DB_NAME', env('DB_NAME'));
-Config::define('DB_USER', env('DB_USER'));
-Config::define('DB_PASSWORD', env('DB_PASSWORD'));
+Config::define('DB_NAME', env('DB_NAME') ?? null);
```

Comment:
`?? null` seems unnecessary.

Reply:
It was done out of @tristanbes's suggestion above to change the Docblock to include `# When using a DSN, you can **remove** DB_NAME, DB_USER, and DB_PASSWORD` instead of leaving it blank. If there isn't an `isset()` on the env variables and you remove them, dotenv will throw an error.

Should I revert it and just assume people won't comment out or remove the DB_ vars?

Reply:
`env()` isn't part of `DotEnv`; it won't throw an error. I'm pretty sure you should be able to remove `?? null` here and it will work just fine. I believe `env()` will return an empty string if the values are in the `.env` file, otherwise it will return null. As for `DotEnv`, it won't throw an error because it only requires them if `DATABASE_URL` isn't in your `.env` (application.php#L31-33).

Feel free to double-check if you want. Or I can check tomorrow.

Reply:
Yeah you're right. I had tested it without uncommenting `DATABASE_URL` and mistakened the ValidationException. It works fine without `?? null` as long as `DATABASE_URL` is set.
```
Fatal error: Uncaught Dotenv\Exception\ValidationException: One or more environment variables failed assertions: DB_NAME is missing, DB_USER is missing, DB_PASSWORD is missing.
```